core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -119,6 +119,26 @@ private CatalogProperties() {}
       "client.pool.cache.eviction-interval-ms";
   public static final long CLIENT_POOL_CACHE_EVICTION_INTERVAL_MS_DEFAULT =
       TimeUnit.MINUTES.toMillis(5);
+  /**
+   * A comma separated list of elements that are used to compose the key of the client pool cache.
+   *
+   * <p>The following elements are supported:
+   *
+   * <ul>
+   *   <li>URI - as specified by {@link CatalogProperties#URI}. URI will be the only element when
+   *       this property is not set.
+   *   <li>UGI - the Hadoop UserGroupInformation instance that represents the current user using the
+   *       cache.
+   *   <li>USER_NAME - similar to UGI but only includes the user's name determined by
+   *       UserGroupInformation#getUserName.
+   *   <li>CONF - name of an arbitrary configuration. The value of the configuration will be
```

Comment:
Can we also add a example what the user can pass in here?

Also, what do you think about make the list here show lower-case, to match what you wrote (ie, there is CONF but then you mention conf.a.b.c)

Reply:
OK. The implementation is actually case-insensitive. I don't have a strong opinion whether to use upper or lower case.